src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -309,10 +309,13 @@ SelectPartsDecision MergeTreeDataMergerMutator::selectPartsToMerge(
                 data_settings->merge_with_ttl_timeout,
                 data_settings->ttl_only_drop_parts);
 
-        parts_to_merge = delete_ttl_selector.select(parts_ranges, max_total_size_to_merge);
+        parts_to_merge = delete_ttl_selector.select(
+            parts_ranges,
+            data_settings->ttl_only_drop_parts ? data_settings->max_bytes_to_merge_at_max_space_in_pool : max_total_size_to_merge);
```

Comment:
Is `max_bytes_to_merge_at_max_space_in_pool` really suitable here? It's a max size for all threads in background pool, shouldn't we divide it between threads here?

Reply:
According to doc, it relates to the number of threads `Maximum in total size of parts to merge, when there are maximum free threads in background pool (or entries in replication queue).`

And in another place, we spread the value among the workers.

https://github.com/ClickHouse/ClickHouse/blob/74a1de564e7add52b27a92c3bb6b3550d67f08f8/src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp#L101-L104

But if, as you said, it does not affect disk space, we can leave it as is.

By the way, can we pass another special value here, e.g., `0` or `uint max`?